service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala

```diff
@@ -114,68 +207,40 @@ case class DefaultCurator(protected val requestModelValidator: CuratorRequestMod
     val parallelServiceExecutor = mahaService.getParallelServiceExecutor(mahaRequestContext)
     val parRequestLabel = "processDefaultCurator"
 
-    val requestModelResultTry = mahaService.generateRequestModel(mahaRequestContext.registryName
-      , mahaRequestContext.reportingRequest
-      , mahaRequestContext.bucketParams)
+    val requestModelResultEither = runAndValidateRequestModel(mahaRequestContext, mahaService, mahaRequestLogBuilder, curatorConfig)
 
-    if(requestModelResultTry.isFailure) {
-      val message = requestModelResultTry.failed.get.getMessage
-      mahaRequestLogBuilder.logFailed(message, Some(400))
-      withError(curatorConfig,
-        GeneralError.from(parRequestLabel
-          , message, new MahaServiceBadRequestException(message, requestModelResultTry.failed.toOption))
-      )
+    if(requestModelResultEither.isLeft) {
+      withError(curatorConfig, requestModelResultEither.left.get)
     } else {
       try {
-        val requestModelResult = requestModelResultTry.get
+        val requestModelResult: RequestModelResult = requestModelResultEither.right.get
         requestModelValidator.validate(mahaRequestContext, requestModelResult)
         val parRequestResult: ParRequestResult = mahaService.executeRequestModelResult(mahaRequestContext.registryName
           , requestModelResult, mahaRequestLogBuilder)
 
+        val (unsortedRequestModelResult, unsortedParRequestResult) =
+          generateUnsortedResults(mahaRequestContext, mahaService, mahaRequestLogBuilder, curatorConfig, requestModelResult, parRequestResult, parRequestResult.queryPipeline.get)
```

Comment:
we are already executing the requestModelResult at line no 218? 
and generateUnsortedResults executing it again with constraints ? 

can you avoid double execution, looks like not right.

Reply:
Per conversation, I will be converting this change into 2 PRs.  First will be only row count change, second will be only code refactor.  Also, yes this is a double execution that is unnecessary.